python/hail/docs/hailpedia/matrixtable.rst

```diff
@@ -0,0 +1,404 @@
+on the :class:`.MatrixTable` object. For example, if the matrix table `mt` had a
+row field `locus`, this field could be referenced with either ``mt.locus`` or
+``mt['locus']``. The former access pattern does not work with field names with
+spaces or punctuation.
+
+The result of referencing a field from a matrix table is an :class:`.Expression`
+which knows its type and knows its source as well as whether it is a row field,
+column field, entry field, or global field. Hail uses this context to know which
+operations are allowed for a given expression.
+
+When evaluated in a Python interpreter, we can see ``mt.locus`` is a
+:class:`.LocusExpression` with type ``locus<GRCh37>`` and it is a row field of
+the MatrixTable `mt`.
+
+    >>> mt
+    <hail.matrixtable.MatrixTable at 0x1107e54a8>
+
+    >>> mt.locus
+    <LocusExpression of type locus<GRCh37>>
+
+Likewise, ``mt.DP`` would be an :class:`.Int32Expression` with type ``int32``
+and is an entry field of `mt`. It is indexed by both rows and columns as denoted
+by its indices when describing the expression:
+
+    >>> mt.DP.describe()
+    --------------------------------------------------------
+    Type:
+        int32
+    --------------------------------------------------------
+    Source:
+        <class 'hail.matrixtable.MatrixTable'>
+    Index:
+        ['row', 'column']
+    --------------------------------------------------------
+
+Import
+======
+
+Text files may be imported with :func:`.import_matrix_table`. Additionally, Hail
+provides functions to import genetic datasets as matrix tables from a
+variety of file formats: :func:`.import_vcf`, :func:`.import_plink`,
+:func:`.import_bgen`, and :func:`.import_gen`.
+
+    >>> mt = hl.import_vcf('data/sample.vcf.bgz')
+
+The :meth:`.MatrixTable.describe` method prints all fields in the table and
+their types, as well as the keys.
+
+    >>> mt.describe()
+    ----------------------------------------
+    Global fields:
+        None
+    ----------------------------------------
+    Column fields:
+        's': str
+    ----------------------------------------
+    Row fields:
+        'locus': locus<GRCh37>
+        'alleles': array<str>
+        'rsid': str
+        'qual': float64
+        'filters': set<str>
+        'info': struct {
+            NEGATIVE_TRAIN_SITE: bool,
+            AC: array<int32>,
+            ...
+            DS: bool
+        }
+    ----------------------------------------
+    Entry fields:
+        'GT': call
+        'AD': array<int32>
+        'DP': int32
+        'GQ': int32
+        'PL': array<int32>
+    ----------------------------------------
+    Column key:
+        's': str
+    Row key:
+        'locus': locus<GRCh37>
+        'alleles': array<str>
+    Partition key:
+        'locus': locus<GRCh37>
+    ----------------------------------------
+
+Common Operations
+=================
+
+Like tables, Hail provides a number of useful methods for manipulating data in a
```

Comment:
I tend to avoid words like "useful" since they don't actually convey any, well, useful information :P